### PR TITLE
[BugFix] reset AsyncRefreshContext when alter mv refreshScheme

### LINKED ISSUE #26962: Modify mv refreshScheme to  ASYNC not work
`alter MATERIALIZED view  mv2 REFRESH ASYNC` not work.

### Steps to reproduce the behavior (Required)
1. CREATE TABLE 
```
CREATE TABLE `event1` (
  `event_id` int(11) NULL COMMENT "Unique query id",
  `event_type` varchar(65533) NULL COMMENT "Unique query id",
  `event_time` datetime NULL COMMENT "Query start time"
) ENGINE=OLAP
DUPLICATE KEY(`event_id`, `event_type`)
COMMENT "OLAP"
PARTITION BY RANGE(`event_time`)
(PARTITION p20230101 VALUES [("2023-01-01 00:00:00"), ("2023-01-02 00:00:00")),
PARTITION p20230102 VALUES [("2023-01-02 00:00:00"), ("2023-01-03 00:00:00")),
PARTITION p20230103 VALUES [("2023-01-03 00:00:00"), ("2023-01-04 00:00:00")),
PARTITION p20230104 VALUES [("2023-01-04 00:00:00"), ("2023-01-05 00:00:00")),
PARTITION p20230105 VALUES [("2023-01-05 00:00:00"), ("2023-01-06 00:00:00")),
PARTITION p20230106 VALUES [("2023-01-06 00:00:00"), ("2023-01-07 00:00:00")))
DISTRIBUTED BY HASH(`event_id`) BUCKETS 1
PROPERTIES (
"replication_num" = "1",
"dynamic_partition.enable" = "false",
"dynamic_partition.time_unit" = "DAY",
"dynamic_partition.time_zone" = "Asia/Shanghai",
"dynamic_partition.start" = "-3",
"dynamic_partition.end" = "3",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "1",
"dynamic_partition.history_partition_num" = "0",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false",
"compression" = "LZ4"
);

CREATE MATERIALIZED VIEW `event1_mv`
COMMENT "MATERIALIZED_VIEW"
PARTITION BY (`event_time`)
DISTRIBUTED BY HASH(`event_id`) BUCKETS 1
REFRESH ASYNC EVERY(INTERVAL 1 DAY)
PROPERTIES (
"replication_num" = "1",
"storage_medium" = "HDD",
"partition_refresh_number" = "1",
"force_external_table_query_rewrite" = "true",
"prefer_compute_node" = "false",
"compute_node_selector" = "",
"partition_ttl_number" = "4"
)
AS SELECT `event1`.`event_id`, `event1`.`event_type`, `event1`.`event_time`
FROM `event_iceberg`.`event1`

```

### Expected behavior (Required)

`show create table event1_mv`  REFRESH not change
### Real behavior (Required)
REFRESH change to  async
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main

### PR DESCRIPTION
**Pull request by hongli-my**
Fixes #26962

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|6|7|85.71%|[568]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|6|7|85.71%|[568]|

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java [hunk 1]
```diff
@@ -559,6 +559,16 @@ private void processChangeRefreshScheme(RefreshSchemeDesc refreshSchemeDesc, Mat
                 }
                 asyncRefreshContext.setStep(step.getLongValue());
                 asyncRefreshContext.setTimeUnit(intervalLiteral.getUnitIdentifier().getDescription());
+            } else {
+                if (materializedView.getBaseTableInfos().stream().anyMatch(tableInfo -> {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java:563]: how about extract this check into the `MaterializedView` ?

Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java:563]: +1，  could be a method

Comment (hongli-my) [reply fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java]: getTable  method  has been implemented in BaseTableInfo

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|4|5|80.00%|[566]|

**Review by murphyatwork — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**